Userland/Libraries/LibThread/Lock.h

```diff
@@ -24,7 +27,22 @@ class Lock {
     void unlock();
 
 private:
-    Atomic<pid_t> m_holder { 0 };
+#ifdef __serenity__
+    using ThreadID = int;
+
+    ALWAYS_INLINE static ThreadID self()
+    {
+        return gettid();
```

Comment:
Maybe we should just use `pthread_t` and `pthread_self()` for SerenityOS. It's pretty much an alias for `gettid()` anyway.